cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -89,24 +92,31 @@ public void getProfiles(HttpRequest request, HttpResponder responder,
   @Path("/profiles/{profile-name}")
   public void getProfile(HttpRequest request, HttpResponder responder,
                          @PathParam("namespace-id") String namespaceId,
-                         @PathParam("profile-name") String profileName) throws NotFoundException {
-    // TODO: implement the method and remove the mock data
-    if (new NamespaceId(namespaceId).equals(NamespaceId.SYSTEM)) {
-      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(SYSTEM_PROFILE));
-    } else {
-      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(USER_PROFILE));
-    }
+                         @PathParam("profile-name") String profileName) throws NotFoundException, IOException {
+    ProfileId profileId = new ProfileId(namespaceId, profileName);
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(profileStore.getProfile(profileId)));
   }
 
   /**
    * Write a profile in a namespace.
    */
   @PUT
   @Path("/profiles/{profile-name}")
-  public void writeProfile(HttpRequest request, HttpResponder responder,
-                           @PathParam("namespace-id") String namespaceId,
-                           @PathParam("profile-name") String profileName) throws BadRequestException {
-    // TODO: implement the method
+  public void writeProfile(
+    FullHttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespaceId,
+    @PathParam("profile-name") String profileName) throws BadRequestException, IOException, AlreadyExistsException {
+    ProfileInfo profileInfo;
+    try (Reader reader = new InputStreamReader(new ByteBufInputStream(request.content()), StandardCharsets.UTF_8)) {
+      profileInfo = GSON.fromJson(reader, ProfileInfo.class);
```

Comment:
you should also validate that the profileInfo is valid, throwing BadRequestException if not. You should be using ProvisioningService.validateProperties() to check that the provisioner exists and its properties are valid.